packages/worker/src/core.ts

```diff
@@ -34,12 +45,16 @@ function defaultTelemetryOptions(): TelemetryOptions {
 export class Core {
   /** Track the registered workers to automatically shutdown when all have been deregistered */
   protected registeredWorkers = new Set<native.Worker>();
+  protected shouldPollForLogs = new BehaviorSubject<boolean>(false);
+  protected logPollPromise = Promise.resolve();
```

Comment:
We don't want a resolved promise as the default, I'd say that the default is a promise that never resolves.
I'd initialize this in the constructor:

```ts
this.logPollPromise = this.initLogPolling(...);
```